### ISSUE TITLE
Not Detecting Internet Activity

### ISSUE THREAD

**Issue by monstercatyy** [bruxo00/steam-auto-shutdown#33]
The new GUI doesnt seem to detect the activity, I am using Wifi and the traffic says 0 and the IDLE starts counting down

**Comment by diogomartino** [comment]
If you use disk monitor, does it detect any activity there?

**Comment by NickBosworth** [comment]
I can confirm the same issue, tested on 2x windows 11 devices. 

There is no disk or network activity detected or displayed at the bottom of the screen. 

Tested all interfaces under normal and run as admin permissions.

If I get a chance ill pull the source and debug on one of those devices to see if we can see what the underlying issue is.

**Comment by diogomartino** [comment]
@NickBosworth if you could debug it, that would be awesome. I have no way of knowing what's happening because for me (and I'm assuming most people) is working fine. Tested just now with both ethernet and wifi and it's working.

![image](https://github.com/diogomartino/steam-auto-shutdown/assets/16502356/be68ceda-76c4-4873-994b-c87afef227e2)

Some guesses:
- It might be possible (but really unlikely) that the frontend cannot attach to the wails runtime to get the values. To check this, open the settings (right corner), in the monitor disk section, click on the icon that's on the side of `steam.exe`. Is there any processes in the modal to pick from?
- If not, the best place to start debugging is `network-manager.go` in the `GetInterfaceDownloadSpeed` function. It should be pretty easy to find the issue, not so sure if it's easy to fix it.

**Comment by monstercatyy** [comment]
During the past 2 weeks I've been using Ethernet for now since my WIFI activity isn't detected and also disk activity is kinda unreliable for me it shutdowns sometimes and waking up to downloads at 50-60%.

**Comment by diogomartino** [comment]
@EmmanuelCy yes, the counters are kinda unreliable. You can try to increase the shutdown timer and the speed threshold in the settings a bit. That might help. About the Wifi not being detected, I can't do much because I can't reproduce the issue, I would need someone that has that problem to debug it on their machine.

**Comment by monstercatyy** [comment]
@diogomartino, I've encountered another bug where no activity is detected on both Network and Disk on all interfaces. Sometimes a restart resolves the issue, but not always. I would revert to the older version, but the max Idle Threshold is insufficient for my needs.

I'd like to debug this myself and try to find the issue, since you guys can't seem to reproduce the bug on your end.

**Comment by jusparr** [comment]
I can also confirm it does nothing for me on the latest win 11

**Comment by LS-FCEFyN** [comment]
On Windows 10 Professional, the same error occurs. However, relaunching the app with admin privileges allows the app to detect network activity. This workaround was only needed once; during subsequent uses of the app, it was not needed anymore.

*Disk activity worked fine with and without admin privileges

I don't really know what sort of information would be useful to you to debug this but I'll try to provide anything I can.

**Comment by nzdus** [comment]
I think i might have found a fix for this issue. For me it was because it was detecting the "Microsoft Wifi Direct Virtual Adapter" and displaying it as "Local Area Connection 2" which wasn't receiving or sending any data. I had to go into device manaver, click on view, show hidden devices, than under network it should be there to disable. I did a restart to be safe, and it now notices my steam downloads with both hdd and network reads. I'll check to see if it picks up on a download from anything else. 

Edit: I have checked with my browser and a download manager for browser downloads and it notices both. I think this may be the solution to the issue.

**Comment by diogomartino** [comment]
@nzdus thanks for that, that might help some people. Just a question, your correct network adapter was not showing in the list for you to pick instead of your that "Microsoft Wifi Direct Virtual Adapter"? And after you disable that adapter, did it detect the correct adapter correctly on startup?

**Comment by nzdus** [comment]
@diogomartino  My correct adapter and the "Local Area Connection 2" were selected at the same time (but the local area connection would take priority). Like they were connected somehow, and yes It displayed the correct adapter on startup after disabling. Even before I disabled it it would show it on startup, but than it would switch to the other one within 30 seconds to a minute.

**Comment by SanCigo** [comment]
I had same problem, windows 11. 

Fixed running as administrator, so it may be a permissions issue (?)

**Comment by KafkaOnTheMoon** [comment]
I also have the same problem, although I fixed it by turning Steam back on (I turned it off earlier), although what I'm installing has nothing to do with Steam. I guess it's not called "Steam auto shutdown" for no reason 🤣🤣.

**Comment by RPullen1** [comment]
I have started to receive this issue recently, the app just doesn't detect movement on the network or disk. I have just installed 4.0 which is working perfectly and detecting the download and install correctly. I'm wondering if the 6.0 issue is related to the Window's Defender warning I get when I load it up, attached screenshot.

Any fixes?
![image](https://github.com/user-attachments/assets/30c89415-43ee-411f-a098-70d65ea9450b)

**Comment by Sawa-i** [comment]
Ah... @diogomartino .. I'd like to add as well.

So when I downloaded it originally, I was updating with steam and it worked fine.

2 weeks later (today), I was updating Wuthering waves (with executable titled "launcher_main.exe")

<img width="782" height="155" alt="Image" src="https://github.com/user-attachments/assets/4ee7f241-fcf5-4f08-9663-0686495b0321" />

And ofcourse I was being shown 0 MBps on the app, moreover I was having trouble identifying where application is selected for tracking. The following I thought meant to "Executable which you want to track for Monitor Disk" (Confusing UI).

<img width="596" height="430" alt="Image" src="https://github.com/user-attachments/assets/8d085fed-60b0-48e2-b8d8-b8dd0f6fc080" />

I was lucky to tap on it randomly and noticed that it opens the " Executable Files running on the background" rather than the "Process name" shown by task manager, i.e.:



<img width="1031" height="323" alt="Image" src="https://github.com/user-attachments/assets/b338b466-8037-4d5a-aa79-4e4672e52ac5" />

<img width="1092" height="328" alt="Image" src="https://github.com/user-attachments/assets/75aef437-c87e-4615-b00f-5b07e568a371" />

<img width="604" height="431" alt="Image" src="https://github.com/user-attachments/assets/6bf18d4c-92cc-4fcc-879d-0785ba714c73" />

So yes, after selecting the app was able to track the Network speed.

Though While writing this comment here, I noticed a bug I believe as I tried changing the app to other with less Networking but I was getting the same Networking. It still happens after reopening the app.

I'm using Windows 11

_______________________________

**Conclusive Remarks:**
1) I believe the Selection of "Executable File" for Tracking should happen in Main Page rather than Settings, Plus it's overshadowed.
1.2) Better if " Process Name"  instead of " Executable Name" , as Chrome can have 9+ executables etc, and so does steam have many own.

2) [BUG] Tracking remained the same even when changed the app for tracking.
